pkg/guacrest/server/server.go

```diff
@@ -99,3 +99,63 @@ func (s *DefaultServer) AnalyzeDependencies(ctx context.Context, request gen.Ana
 		return nil, fmt.Errorf("%v sort is unsupported", request.Params.Sort)
 	}
 }
+
+func (s *DefaultServer) GetPackagePurls(ctx context.Context, request gen.GetPackagePurlsRequestObject) (gen.GetPackagePurlsResponseObject, error) {
+	return gen.GetPackagePurls200JSONResponse{}, nil
+}
+
+func (s *DefaultServer) GetPackageVulns(ctx context.Context, request gen.GetPackageVulnsRequestObject) (gen.GetPackageVulnsResponseObject, error) {
+	return gen.GetPackageVulns200JSONResponse{}, nil
+}
+
+func (s *DefaultServer) GetPackageDeps(ctx context.Context, request gen.GetPackageDepsRequestObject) (gen.GetPackageDepsResponseObject, error) {
+	purls, err := GetDepsForPackage(ctx, s.gqlClient, request.Purl)
```

Comment:
The purl should be unescaped in the endpoint handlers (here, in `server.go`) instead of in the business logic (`retrieveDependencies.go`) because it is something specific to this REST API and independent of the retrieval logic. This will allow using `retrieveDependencies.go` as a library in other places that don't require unescaping the input or unescaping it in the same way. 

This also simplifies testing - a simple test can be added to `server.GetPackageDeps` to verify that the unescaping is done correctly, and the tests for the functions in `retrieveDependencies` don't need to worry about encoding.

Reply:
Done, thanks. I didn't think about doing the unescaping in `server.go`.